object-manipulation-2/src/pick.js

```diff
@@ -0,0 +1,16 @@
+/* eslint-disable no-unused-vars */
+
+function pick(source, keys) {
+  var result = {};
+  for (var propertyNames in source) {
+    if (keys.includes(propertyNames)) {
+      result[propertyNames] = source[propertyNames];
+    }
```

Comment:
You can add another predicate to your `if` condition. `&& source[propertyName] !== undefined`. Then you don't need the other loop.